### DISCUSSION TITLE
Roles not assigned?

### DISCUSSION THREAD

**Discussion by DevTiago** [spatie/laravel-permission#1828] [Q&A] [answered]
Greetings,

First of all, thanks for the package!

I run my migrations and im using just roles (no permissions so far) and i have the user, role and model_has_role tabes, filled correctly.

On User model i have this:

```
class User extends Authenticatable implements JWTSubject
{
    use Notifiable, HasRoles, Rateable, Impersonate;
```

but when i try to check the role from user always return false:

`$user->hasRole('Admin')`

Even if i try this to get all users with a certain role only return an empty collection:

`User::role('Admin')->get()`

I think the package is correctly installed because if i try to get info from a role that doesn't exist it returns the right message:

```
Spatie\Permission\Exceptions\RoleDoesNotExist
There is no role named `writer`.
```

Any advice?

**Comment by erikn69** [comment]
`JWTSubject`? have you another guard?
Did you try `$user->hasRole('Admin','web')` or `$user->hasRole('Admin','api')`( for example)?
Have you a custom Role model with hidden `guard_name`?
Laravel version? package version?

**Comment by devkuros** [comment]
@erikn69 hi.. i have the same problem.. the problem is, i try to put the different name for guard_name.

1. admin with web guard_name, and
2. customer with customer guard_name

when i tried to create user with assignRole customer is error with message

```
Spatie\Permission\Exceptions\RoleDoesNotExist
There is no role named 'customer'.
```

and everthing roled name with web guard_name got no errors

**Reply by erikn69** [reply]
try release 4.4.0

**Comment by DevTiago** [comment, marked as answer]
@devkuros be careful with uppercase letters, the role 'Customer' could not be the same as the role 'customer'.

My problem was solved: on table '**model_has_roles**' the _model_type_ field was `App\User` but i've updated my project to Laravel 8 and the models are now on a Models folder. So, the model_type has to be `App\Models\User` so the relation could exist.

Thanks for your time!
